libs/ansible/src/hooks/useFetch.ts

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import { deleteData, fetchData, patchData, postData } from '../utils/apiCalls';
+import { PatchRequest } from '@flightctl/types';
+
+export const useFetch = (getCookie: (name: string) => string | undefined, serviceUrl = '') => {
+  const applyHeaders = React.useCallback(
+    (options: RequestInit) => {
+      const token = getCookie('csrftoken');
+      if (token) {
+        if (options.headers) {
+          options.headers['X-CSRFToken'] = token;
+        } else {
+          options.headers = { 'X-CSRFToken': token };
+        }
+      }
+      return options;
+    },
+    [getCookie],
+  );
+
+  const get = React.useCallback(
+    async <R>(kind: string, abortSignal?: AbortSignal): Promise<R> => fetchData(kind, serviceUrl, abortSignal),
+    [serviceUrl],
+  );
+
+  const post = React.useCallback(
+    async <R>(kind: string, obj: R): Promise<R> => postData(kind, obj, serviceUrl, applyHeaders),
+    [serviceUrl, applyHeaders],
+  );
+
+  const remove = React.useCallback(
+    async <R>(kind: string, abortSignal?: AbortSignal): Promise<R> =>
+      deleteData(kind, serviceUrl, applyHeaders, abortSignal),
+    [serviceUrl, applyHeaders],
+  );
+
+  const patch = React.useCallback(
+    async <R>(kind: string, obj: PatchRequest, abortSignal?: AbortSignal): Promise<R> =>
+      patchData(kind, obj, serviceUrl, applyHeaders, abortSignal),
+    [serviceUrl, applyHeaders],
+  );
+
+  const getWsEndpoint = React.useCallback(
+    () => ({
+      wsEndpoint: '',
```

Comment:
no ws yet

Reply:
Should we avoid showing the Terminal tab in that case?

Reply:
yep, i've added a prop to hide the tab and it is being used from ansible lib